### DISCUSSION TITLE
how to deactivate the keymaps, "s" and "S", in visual (x) mode?

### DISCUSSION THREAD

**Discussion by hopezh** [ggandor/leap.nvim#146] [Q&A] [answered]
Leap.nvim can be used in the visual mode (i.e. the x mode) with "s" and "S". 

I want to deactivate this keymap so that I can use "s"/"S" to add surrounding symbols to a visual selection as offered by nvim-surround. 

May I ask how to do this by editing leap.nvim's keymap configurations  (I'm using LazyVim)? 

Thanks.

**Comment by Praczet** [comment]
Hi,

I think by the default in the LazyVim configuration you can surround using motion `gza`

`gz` is surround when `a` is add and then you choose with what you surround.

**Reply by hopezh** [reply]
Thanks for the tip, and it works. 

If I do want to remove the keymap of "s" and "S" in visual mode for leap.nvim, what should I do?

**Reply by Praczet** [reply]
To remove keymap you can:

```lua
vim.keymap.del({ "x" }, "s")
vim.keymap.del({ "x" }, "S")
```

 but this remove mapping for the characters: `s,S` after that you should add mapping for surrounding.

**Reply by Praczet** [reply, marked as answer]
In the [LazyVim's webpage](https://www.lazyvim.org/plugins/editor) you can check how leap.nvim is configured.

```lua
{
  "ggandor/leap.nvim",
  keys = {
    { "s", mode = { "n", "x", "o" }, desc = "Leap forward to" },
    { "S", mode = { "n", "x", "o" }, desc = "Leap backward to" },
    { "gs", mode = { "n", "x", "o" }, desc = "Leap from windows" },
  },
  config = function(_, opts)
    local leap = require("leap")
    for k, v in pairs(opts) do
      leap.opts[k] = v
    end
    leap.add_default_mappings(true)
    vim.keymap.del({ "x", "o" }, "x")
    vim.keymap.del({ "x", "o" }, "X")
  end,
}
```

**Reply by hopezh** [reply]
Thanks. I've removed s, S from the visual model ("x") from leap.nvim's default configs in a new leap.lua file in lazyvim's plugin folder:

```
    ...
    keys = {
      { "s", mode = { "n", "o" }, desc = "Leap forward to" },
      { "S", mode = { "n", "o" }, desc = "Leap backward to" },
      { "gs", mode = { "n", "o" }, desc = "Leap from windows" },
    },
    ...
```

**Comment by unphased** [comment]
I'm finding that the behavior we get is that leap.nvim does not override surround's maps (when initializing it afterward) since it comes with code to check them before assigning and prints the warning message. I guess having the warning message clutter up output is not ideal, but I am leaning towards just leaving it like this. Because it seems kind of complicated to set this right. I made this config in plugins:

```
  {
    "ggandor/leap.nvim",
    keys = {
      { "s", mode = { "n", "o" }, desc = "Leap forward to" },
      { "S", mode = { "n", "o" }, desc = "Leap backward to" },
      { "gs", mode = { "n", "o" }, desc = "Leap from windows" },
    },
  },
```

And here is my init.lua

```
require('leap').create_default_mappings()
-- Or just set to grey directly, e.g. { fg = '#777777' },
-- if Comment is saturated.
vim.api.nvim_set_hl(0, 'LeapBackdrop', { link = 'Comment' })
```

warning still shows up about it trying to assign a 'x' mapping for 'S'.

**Reply by unphased** [reply]
It always seems so misleading when `keys` are shown in examples but modifying it wont ever do anything. But i guess thats a problem to take up with Lazy project.
